src/cache/redis.ts

```diff
@@ -111,52 +111,68 @@ export class RedisCache<T = unknown> implements Cache<T> {
     recordRedisCommandMetric(CMD_SENT_STATUS.SUCCESS, 'exec')
   }
 
-  async lock(key: string, cacheLockDuration: number, retryCount: number): Promise<void> {
+  async lock(
+    key: string,
+    cacheLockDuration: number,
+    retryCount: number,
+    shutdownNotifier: EventEmitter,
+  ): Promise<void> {
     const start = Date.now()
     const log = (msg: string) => `[${(Date.now() - start) / 1000}]: ${msg}`
-    const durationBuffer = cacheLockDuration * 0.7
+    // Ensures lock is able to be extended in time
+    const durationBuffer = cacheLockDuration * 0.8
```

Comment:
Let's perform this calculation in the extension loop inline, we can explain it with a comment and it'll clearer there imo